sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkSyncHttpResponse.java

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.jdk.httpclient;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.WritableByteChannel;
+
+import static com.azure.core.http.jdk.httpclient.JdkHttpClient.fromJdkHttpHeaders;
+
+final class JdkSyncHttpResponse extends JdkHttpResponseBase {
+    private static final ClientLogger LOGGER = new ClientLogger(JdkSyncHttpResponse.class);
+    private BinaryData binaryData = null;
+    public static final int STREAM_READ_SIZE = 8192;
+
+    private final InputStream bodyStream;
+    private byte[] bodyBytes;
+
+    private volatile boolean disposed = false;
+    JdkSyncHttpResponse(final HttpRequest request, int statusCode, HttpHeaders headers, byte[] bytes) {
+        super(request, statusCode, headers);
+        this.bodyStream = null;
+        this.bodyBytes = bytes;
+    }
+
+    JdkSyncHttpResponse(final HttpRequest request, java.net.http.HttpResponse<InputStream> streamResponse) {
+        super(request, streamResponse.statusCode(), fromJdkHttpHeaders(streamResponse.headers()));
+        this.bodyStream = streamResponse.body();
+        this.bodyBytes = null;
+    }
+
+    @Override
+    public Flux<ByteBuffer> getBody() {
+        if (bodyBytes != null) {
+            return Mono.fromSupplier(() -> ByteBuffer.wrap(bodyBytes)).flux();
+        } else {
+            return FluxUtil.toFluxByteBuffer(bodyStream).doFinally(ignored -> close());
+        }
+    }
+
+    @Override
+    public Mono<byte[]> getBodyAsByteArray() {
+        if (bodyBytes != null) {
+            return Mono.just(bodyBytes);
+        } else {
+            return super.getBodyAsByteArray();
+        }
+    }
+
+    @Override
+    public BinaryData getBodyAsBinaryData() {
+        if (bodyBytes != null) {
+            return BinaryData.fromBytes(bodyBytes);
+        } else {
+            // we shouldn't create multiple binary data instances for a single stream
```

Comment:
just curious, what happens if we do?

Reply:
Imagine we do this:

data1 = response.getBodyAsBinaryData();
bytes1 = data.toBytes();

Reply:
Imagine we do this:

```java
data1 = response.getBodyAsBinaryData();
bytes1 = data1.toBytes();

data2 = response.getBodyAsBinaryData();
bytes2 = data2.toBytes(); 
```

Since there is just one stream underneath, `data1.toBytes()` will read the stream to the end and `data2.toBytes()` will have nothing to read and return an empty array.

Assuming we read data1 partially, then data2 will read the rest, i.e. sharing binary data over one stream can result in complicated and non-desirable behavior.